Android Source/mobile/src/main/java/net/steppschuh/sensordatalogger/PhoneActivity.java

```diff
@@ -117,6 +129,54 @@ public void onClick(View v) {
             }
         });
 
+        floatingActionRecordButton = (FloatingActionButton) findViewById(R.id.recordActionButton);
+        floatingActionRecordButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // request runtime permission
+                if (ContextCompat.checkSelfPermission(getApplicationContext(),
```

Comment:
As the minimum SDK version is 21, you can probably avoid requesting this permission. According to the [documentation](https://developer.android.com/reference/android/Manifest.permission#WRITE_EXTERNAL_STORAGE):

> Starting in API level 19, this permission is not required to read/write files in your application-specific directories returned by Context.getExternalFilesDir(String)

Reply:
(1/3) Removed the permission checking, good to know that it was not required. :+1: